tests/models/mt5/test_modeling_mt5.py

```diff
@@ -12,14 +12,1055 @@
+            )
+            random_slice_idx = ids_tensor((1,), model_result[0].shape[-1]).item()
+
+            # check that outputs are equal
+            self.parent.assertTrue(
+                torch.allclose(
+                    model_result[0][0, :, random_slice_idx], tied_model_result[0][0, :, random_slice_idx], atol=1e-4
+                )
+            )
+
+            # check that outputs after saving and loading are equal
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                tied_model.save_pretrained(tmpdirname)
+                tied_model = model_class.from_pretrained(tmpdirname)
+                tied_model.to(torch_device)
+                tied_model.eval()
+
+                # check that models has less parameters
+                self.parent.assertLess(
+                    sum(p.numel() for p in tied_model.parameters()), sum(p.numel() for p in model.parameters())
+                )
+                random_slice_idx = ids_tensor((1,), model_result[0].shape[-1]).item()
+
+                tied_model_result = tied_model(
+                    input_ids=input_ids,
+                    decoder_input_ids=decoder_input_ids,
+                    attention_mask=attention_mask,
+                    decoder_attention_mask=decoder_attention_mask,
+                )
+
+                # check that outputs are equal
+                self.parent.assertTrue(
+                    torch.allclose(
+                        model_result[0][0, :, random_slice_idx],
+                        tied_model_result[0][0, :, random_slice_idx],
+                        atol=1e-4,
+                    )
+                )
+
+    def check_resize_embeddings_t5_v1_1(
+        self,
+        config,
+    ):
+        prev_vocab_size = config.vocab_size
+
+        config.tie_word_embeddings = False
+        model = MT5ForConditionalGeneration(config=config).to(torch_device).eval()
+        model.resize_token_embeddings(prev_vocab_size - 10)
+
+        self.parent.assertEqual(model.get_input_embeddings().weight.shape[0], prev_vocab_size - 10)
+        self.parent.assertEqual(model.get_output_embeddings().weight.shape[0], prev_vocab_size - 10)
+        self.parent.assertEqual(model.config.vocab_size, prev_vocab_size - 10)
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        (
+            config,
+            input_ids,
+            decoder_input_ids,
+            attention_mask,
+            decoder_attention_mask,
+            lm_labels,
+        ) = config_and_inputs
+
+        inputs_dict = {
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+            "decoder_input_ids": decoder_input_ids,
+            "decoder_attention_mask": decoder_attention_mask,
+            "use_cache": False,
+        }
+        return config, inputs_dict
+
+
+@require_torch
```

Comment:
pretty sure we can use some copied from here as well!

Reply:
Sure, I added a "copied from" to this class as well. 

I had to add a new MT5_PRETRAINED_MODEL_ARCHIVE_LIST, which should be fine.

Another issue I ran into is the lack of mT5 in tiny_model_summary.json. The tiny model summary workflow didn't generate a tiny summary for mT5 because there were no tests for mT5:
https://github.com/huggingface/transformers/actions/runs/7705297481/job/20998991348#step:6:139

I will need to merge this change before the tiny model summary workflow generates the summary for mT5, and update the tiny model summary file in a following change. Otherwise, the tokenizer_name is None and the pipeline test for mT5 won't run.

Reply:
perfect. For that specific test, you can add a `# Ignore copy` mention on top of it, and just ignore it no need to run it!

Reply:
Cool, thanks. 

The copied method was "_is_pipeline_test_to_skip_()", and I ended up just added an extra None check for the tokenizer_name in the original method in test_modeling_t5. The None check should cause the test to be skipped for MT5, so shouldn't need the `#Ignore copy` in this case. 

I'll just skip the tiny model summary update then.